module/VuFind/src/VuFind/RecordDriver/EDS.php

```diff
@@ -460,12 +460,13 @@ public function getTitle()
      */
     public function getPrimaryAuthors()
     {
-        return array_unique(
-            $this->extractEbscoDataFromRecordInfo(
-                'BibRecord/BibRelationships/HasContributorRelationships/*/'
+        $authors = $this->extractEbscoDataFromRecordInfo(
+            'BibRecord/BibRelationships/HasContributorRelationships/*/'
                 . 'PersonEntity/Name/NameFull'
-            )
         );
+        return array_unique(array_filter($authors, function ($a) {
```

Comment:
I believe you can simplify this to simply `return array_unique(array_filter($authors));` -- the default behavior of array_filter is to filter out empty elements, so the callback should not be required in this situation. Do you mind giving it a try and updating the PR if it works?

Reply:
Thanks, yes, it works. I have just updated that. As for the placement inside the `getPrimaryAuthors()`, the lower function, `extractEbscoDataFromRecordInfo()`, is used many times in the whole class but this is the only place where its output is deduplicated this way.  Of course, I will be happy with any addition.

Reply:
Great, thank you! I've also reviewed all the uses of `extractEbscoDataFromRecordInfo` and I agree that it currently makes the most sense to leave this fix where you put it. I'll backport this to the release-7.0 branch so the fix can be included in next week's 7.0.2 release.